dspace-api/src/main/java/org/dspace/content/authority/service/ChoiceAuthorityService.java

```diff
@@ -173,4 +172,47 @@ public Choices getBestMatch(String fieldKey, String query, Collection collection
      */
     public void clearCache();
 
+    /**
+     * Should we store the authority key (if any) for such field key and collection?
+     * 
+     * @param fieldKey   single string identifying metadata field
+     * @param collection Collection owner of Item or where the item is submitted to
+     * @return true if the configuration allows to store the authority value
+     */
+    public boolean storeAuthority(String fieldKey, Collection collection);
+
+    /**
+     * Wrapper that calls getChoicesByParent method of the plugin.
+     *
+     * @param authorityName authority name
+     * @param parentId      parent Id
+     * @param start         choice at which to start, 0 is first.
+     * @param limit         maximum number of choices to return, 0 for no limit.
+     * @param locale        explicit localization key if available, or null
+     * @return a Choices object (never null).
+     * @see org.dspace.content.authority.ChoiceAuthority#getChoicesByParent(java.lang.String, java.lang.String,
+     *  int, int, java.lang.String)
+     */
+    public Choices getChoicesByParent(String authorityName, String parentId, int start, int limit, String locale);
+
+    /**
+     * Wrapper that calls getTopChoices method of the plugin.
+     *
+     * @param authorityName authority name
+     * @param start         choice at which to start, 0 is first.
+     * @param limit         maximum number of choices to return, 0 for no limit.
+     * @param locale        explicit localization key if available, or null
+     * @return a Choices object (never null).
+     * @see org.dspace.content.authority.ChoiceAuthority#getTopChoices(java.lang.String, int, int, java.lang.String)
+     */
+    public Choices getTopChoices(String authorityName, int start, int limit, String locale);
+
+    /**
+     *
+     * @param authorityName authority name
+     * @param vocabularyId  child id
+     * @param locale        explicit localization key if available, or null
+     * @return the parent Choice object if any
+     */
+    public Choice getParentChoice(String authorityName, String vocabularyId, String locale);
```

Comment:
Again, please add a description to this JavaDocs

Reply:
https://github.com/DSpace/DSpace/pull/2792/files#diff-52f85bc6d10ca3376c18c0d76e37dd0bR217-R218